src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -189,21 +189,15 @@ def filter(
         RegressionProblem: a regression problem data class
         """
 
-        filtered_rvs = []
+        posterior = FilteringPosterior(transition=self.dynamics_model)
         info_dicts = []
 
-        for rv, info in self.filtered_states_generator(
-            regression_problem, _previous_posterior
+        for idx, (rv, info) in enumerate(
+            self.filtered_states_generator(regression_problem, _previous_posterior)
         ):
-            filtered_rvs.append(rv)
+            posterior.append(location=regression_problem.locations[idx], state=rv)
             info_dicts.append(info)
 
```

Comment:
do you think the generator should yield t as well? This enumerate seems a bit awkward here, doesnt it?

Reply:
Done